### PR TITLE
Articles pagination

### PR DESCRIPTION
**Pull request by tyndria**
Add pagination for main page articles

### CODE REVIEW

File: constants/api.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ export default {
   },
   articles: {
     getAll: `${API_URL}/articles`,
+    getChunk: (page, pageSize) => `${API_URL}/articles?page=${page}&pageSize=${pageSize}`,
```
Comment (Drapegnik) [inline constants/api.js]: * you should modify `getAll`, not create new endpoint
* `page=${page}` what about to use some default solutions (like [`qs`](https://github.com/ljharb/qs) for query params?

File: pages/index.js [hunk 2]
```diff
@@ -36,38 +41,53 @@ class HomePage extends Component {
     }).isRequired,
     articles: ArticlesArray.isRequired,
     diary: DiaryShape.isRequired,
+    pagination: PropTypes.oneOfType([PropTypes.bool, PaginationShape]).isRequired,
```
Comment (Drapegnik) [inline pages/index.js]: `bool`?

File: pages/index.js [hunk 3]
```diff
@@ -36,38 +41,53 @@ class HomePage extends Component {
...
+
+    return request.populate(ctx, initialRequests);
   }
 
   render() {
-    const { articles, error, diary, getByDay, url } = this.props;
+    const { articles, error, diary, pagination, getByDay, getChunk, url } = this.props;
+
+    const articlesRows = getArticlesRows(articles, ROW_SIZE);
+    const firstPageRows = { first: articlesRows[0], second: articlesRows[1] };
```
Comment (Drapegnik) [inline pages/index.js]:
```js

Comment (Drapegnik) [inline pages/index.js]:
```js
const [first, second] = aarticlesRows
```

File: pages/index.js [hunk 4]
```diff
@@ -36,38 +41,53 @@ class HomePage extends Component {
...
+          <div className="page-content">
+            <ArticlesRow articles={firstPageRows.first} className="first-line is-ancestor" />
+            <ArticlesComplexRow
+              articles={firstPageRows.second}
+              renderDiary={() => (
+                <Diary {...diary} getNextDiary={() => getByDay()} getPrevDiary={() => getByDay()} />
+              )}
+            />
+
+            {articles.length > PAGE_SIZE &&
```
Comment (Drapegnik) [inline pages/index.js]: Please extract all this logic to component `Pagination`

Comment (Drapegnik) [reply pages/index.js]: Possible api:
```jsx
  <Pagination
    page={}
	pageSize={}
  />
```

File: redux/ducks/articles.js [hunk 5]
```diff
@@ -36,13 +38,21 @@ const currentReducer = defaultReducer((state, { payload }) => ({
...
 export default createReducer(
   {
     [FETCH_ALL]: defaultReducer((state, { payload: { data } }) => ({
       ...state,
       data,
       pending: false,
     })),
+    [FETCH_CHUNK]: defaultReducer((state, { payload: { data, next } }) => ({
+      ...state,
+      data: next.page === FIRST_PAGE_NUMBER + 1 ? data : [...state.data, ...data],
```
Comment (Drapegnik) [inline redux/ducks/articles.js]: what about init data with `[]` & omit this ternary?

File: utils/getters.js [hunk 6]
```diff
@@ -47,3 +47,11 @@ export const getLocalizedArticle = (article, lang) => {
 
 export const getLocalizedArticles = (articles, lang) =>
   articles.map(article => getLocalizedArticle(article, lang));
+
+export const getArticlesRows = (articles, rowSize) => {
```
Comment (Drapegnik) [inline utils/getters.js]: check out [`lodash@chunk`](https://lodash.com/docs/4.17.4#chunk)

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: constants/api.js (hunk 1 above)
Comment (tyndria) [reply constants/api.js]: @Drapegnik the library is nice, agree, but does it make sense to connect it for a such simple case?

File: pages/index.js (hunk 2 above)
Comment (tyndria) [reply pages/index.js]: It will be ``false`` when there are no more pages. not so obviously, agree(

File: constants/api.js (hunk 1 above)
Comment (Drapegnik) [reply constants/api.js]: It's already used

File: redux/ducks/articles.js (hunk 5 above)
Comment (tyndria) [reply redux/ducks/articles.js]: @Drapegnik , this action is dispatched when you are navigating across different tabs or come to the main page through Back button. In a such case the store isn't reloaded.
I could try to reload it as you do here https://github.com/babajka/babajka-frontend/blob/master/routes.js#L36 or leave code as it is

File: constants/api.js (hunk 1 above)
Comment (tyndria) [reply constants/api.js]: hmm, I see it in package.json, but I can't find where you are using it

File: utils/getters.js (hunk 6 above)
Comment (tyndria) [reply utils/getters.js]: thanks! I will definitely use it

File: redux/ducks/articles.js [hunk 7]
```diff
@@ -69,11 +69,11 @@ export default createReducer(
 export const actions = {
   fetchAll: () => ({
     type: FETCH_ALL,
-    payload: request.fetch(api.articles.getAll),
+    payload: request.fetch(api.articles.getAll()),
   }),
-  fetchChunk: (page, pageSize) => ({
+  fetchChunk: (page = 0, pageSize = 8) => ({
```
Comment (Drapegnik) [inline redux/ducks/articles.js]: use constants

File: constants/api.js [hunk 8]
```diff
@@ -8,8 +8,10 @@ export default {
     logout: `${AUTH_URL}/logout`,
   },
   articles: {
-    getAll: `${API_URL}/articles`,
-    getChunk: (page, pageSize) => `${API_URL}/articles?page=${page}&pageSize=${pageSize}`,
+    getAll: (page, pageSize) => {
+      const baseUrl = `${API_URL}/articles`;
+      return page && pageSize ? `${baseUrl}?page=${page}&pageSize=${pageSize}` : baseUrl;
```
Comment (Drapegnik) [inline constants/api.js]: `page && pageSize ?` - you feel that this bicycle becomes more and more confused?

please use `qs`, or, if you want optimize bundle size, implement your own method `formatUrl(baseUrl, { page, pageSize })`

**Review by Drapegnik — APPROVED** [review]

**Pull request merged**